File: Selectionsort/java/selectionDoubleDecrescente.java
public class selectionDoubleDecrescente {
	/*
	* Descricao: essa funcao troca dois elementos de um vetor double
	* Parametros: um vetor double (vetor que tera os 
	* elementos trocados) e dois inteiros (indice dos 
	* dois elementos a serem trocados)
	*/
	public static void swap (double[] vet, int i, int j) {
		double aux = vet[i];
		vet[i] = vet[j];
		vet[j] = aux;
	}
    
    /* 
	 * Descricao: essa funcao ordena um vetor double
	 * em ordem crescente com o metodo selectionsort
	 * 
	 * Parametro: um vetor double (vetor a ser ordenado)
	 */
	public static void selectionsort (double[] vet) {
		for (int i = 0; i < vet.length-1; i ++) {
			int maior = i;

			for (int j = i + 1; j < vet.length; j++) {
				if (vet[maior] < vet[j]) {
					maior = j;
				}
			}

			if (i != maior) {
				swap(vet, i, maior);
			}
		}
	}
}

File: Selectionsort/c/selectionDoubleDecrescente.c
#include <stdio.h>
#include <stdlib.h>

/*
Descricao: essa funcao troca dois elementos de um vetor de double
Parametros: um vetor de double (vetor que tera os 
elementos trocados) e dois inteiros (indice dos 
dois elementos a serem trocados)
*/
void swap (double vet[], int i, int j) {
	double aux = vet[i];
	vet[i] = vet[j];
	vet[j] = aux;
}

/* 
Descricao: essa funcao ordena um vetor de double
em ordem decrescente com o metodo selectionsort

Parametro: um vetor de double (vetor a ser ordenado)
e um inteiro (tamanho do vetor)
*/
void selectionsort (double vet[], int n) {
	for (int i = 0; i < n-1; i ++) {
		int maior = i;

		for (int j = i + 1; j < n; j++) {
			if (vet[maior] < vet[j]) {
				maior = j;
			}
		}

		if (i != maior) {
			swap(vet, i, maior);
		}
	}
}